blas/gonum/level2double.go

```diff
@@ -168,20 +168,16 @@ func (Implementation) Dger(m, n int, alpha float64, x []float64, incX int, y []f
 		y = y[:n]
 		for i, xv := range x {
 			tmp := alpha * xv
-			if tmp != 0 {
-				atmp := a[i*lda : i*lda+n]
-				f64.AxpyUnitaryTo(atmp, tmp, y, atmp)
-			}
+			atmp := a[i*lda : i*lda+n]
+			f64.AxpyUnitaryTo(atmp, tmp, y, atmp)
```

Comment:
It seems that me that `tmp` could be calculated in line in the call to `f64.AxpyUnitaryTo` since the only reason it existed before was to allow the conditional.

Also, it there a reason not to use `f64.AxpyUnitary` and do the slicing in line too?